core/vm/contract.go

```diff
@@ -143,11 +145,16 @@ func (c *Contract) AsDelegate() *Contract {
 
 // GetOp returns the n'th element in the contract's byte array
 func (c *Contract) GetOp(n uint64) OpCode {
-	if n < uint64(len(c.Code)) {
+	if n < c.CodeEndOffset() {
 		return OpCode(c.Code[n])
 	}
 
-	return STOP
+	// For legacy contracts running out of code section means STOP
+	if c.header.codeSize == 0 {
```

Comment:
I'd say a helper for this would be nice, otherwise just leave a comment stating that based EIP-3540 `codeSize == 0` is invalid, i.e. it must be legacy.